### PR TITLE
Add methods for authenticating with a SSO-Provider

### PR DESCRIPTION
**Pull request by JCWasmx86**
This allows to get a specific SSO-URL for a provider (E.g. https://matrix.org/_matrix/client/v3/login/sso/redirect/oidc-apple?redirectUrl=example.com for Apple), instead of just a generic URL. (Like https://matrix.org/_matrix/client/v3/login/sso/redirect?redirectUrl=example.com)

### CODE REVIEW

**Comment by zecakeh** [comment]
I don't think it makes sense to have two separate methods each time. You should just add the optional param to the current methods. Also I'm wondering if we should just take the id of the `IdentityProvider` as we don't use the rest of the struct.

File: crates/matrix-sdk/src/client.rs [hunk 1]
```diff
@@ -995,6 +1009,10 @@ impl Client {
     ///   associated with the device_id. Only necessary the first time you login
     ///   with this device_id. It can be changed later.
     ///
+    /// * `idp_id` - If None, the returned URL will be a generic one, where one
+    ///   has to choose the provider. If Some, the URL will redirect to the
+    ///   login for the specified provider.
```
Comment (zecakeh) [inline crates/matrix-sdk/src/client.rs:1014]: You should be more generic for this method like:

```suggestion
    /// * `idp_id` - The optional ID of the identity provider to login with.
```

**Comment by zecakeh** [comment]
Don't forget to update the examples in the docs, and the tests at the bottom of the file.

**Comment by zecakeh** [comment]
Can you rebase to make CI happy?

**Comment by JCWasmx86** [comment]
Is there still any issue with this? Or can this be approved+merged?

**Review by poljar — APPROVED** [review]
Looks good.

**Pull request merged**